lib/galaxy/config/sample/datatypes_conf.xml.sample

```diff
@@ -861,6 +861,7 @@
     <datatype extension="ffindex" type="galaxy.datatypes.tabular:Tabular" display_in_upload="true" subclass="true" description_url="https://github.com/soedinglab/ffindex_soedinglab"/>
     <datatype extension="ffdata" type="galaxy.datatypes.data:Data" display_in_upload="true" subclass="true" description_url="https://github.com/soedinglab/ffindex_soedinglab"/>
     <datatype extension="pretext" type="galaxy.datatypes.binary:Pretext" display_in_upload="true" />
+    <datatype extension="iml_dataset" type="galaxy.datatypes.text:Html" subclass="True"/>
```

Comment:
I would prefer a different extension here. `_dataset` seems unintuitive. Maybe `iml_summary`?

Reply:
We need a name to signify that it is more than a summary. I guess _dataset is sort of redundant. Maybe `immuneml_data`?

Reply:
The tools: https://github.com/uio-bmi/immuneml_tools/tree/iml_dataset

Reply:
This one: 
https://github.com/uio-bmi/immuneml_tools/blob/iml_dataset/immuneml_simulate_events.xml
both consumes and creates an iml_dataset

Reply:
A logical sequence for a session could be
 simulate_dataset -> simulate_events -> train_ml_model -> apply_ml_model

Reply:
The tools: https://github.com/uio-bmi/immuneml_tools/tree/iml_dataset

This one:
https://github.com/uio-bmi/immuneml_tools/blob/iml_dataset/immuneml_simulate_events.xml
both consumes and creates an iml_dataset

A logical sequence for a session could be
simulate_dataset -> simulate_events -> train_ml_model -> apply_ml_model

Reply:
To clarify: What is consumed by the next tool is a full ImmuneML dataset. And this consists of multiple files under the extra_files_path. Hence, the html serves both as a summary of the full dataset, as well as a way to pass the full dataset onwards.

Reply:
Then you need to define this as composite datatype (see https://docs.galaxyproject.org/en/master/dev/data_types.html#defining-basic-composite-datatypes), or?

Reply:
@bernt-matthias Yes, I agree, which is sort of the point. However due to time constraints (GCC2021 coming up quite soon...) and since HTML is a composite datatype, we decided to just subclass it for now in the `datatypes_conf.xml` file. Another argument for this solution is to make it easy for Galaxy admins to install the tools without having to update Galaxy (or the dataset package); they can instead just add a line to the `datatypes_conf.xml` file manually.

There will probably be a need to write a separate datatype class at some point, but for now HTML provides what we need. The only problem with using HTML directly is that we do not want just any HTML dataset to be selectable from the tool as input.

Reply:
@bgruening I don't think the naming of the datatype is of importance to us, so we can use whatever you think looks good. I have now asked the devs to suggest a name that more clearly describes the contents.